docs/development-guide.md

```diff
@@ -122,13 +122,31 @@ Signed-off-by: Random J Developer <random@developer.example.org>
 The format can be described more formally as follows:
 
 ```text
-<subsystem>: <what changed>
+<component>: <subject of the change>
 <BLANK LINE>
-<why this change was made>
+<paragraph(s) with reason/description>
 <BLANK LINE>
-<footer>
+<signed-off-by>
 ```
 
+The `component` in the subject of the commit message can be one of the following:
```

Comment:
@nixpanic its also good to add a tag called, dependency or vendor.

Reply:
This is caught by `rebase` as dependencies or vendor updates normally update versions. New dependencies can also be introduced with `build` if it is a new direct dependency of ceph-csi.

Would that cover things sufficiently?

Reply:
Yeah, indirectly `rebase` can include it. To be explicit, that tag helps.. but I am fine with either way. Let me know your choice, based on that we can go ahead and approve.

Reply:
I prefer to stick with `rebase` for now. If we later on notice that `vendor` is something we would like to add, we can always do so.